File: LinkedList/fanzhuanlianbiao.cpp
#include <iostream>
using namespace std;

// é“¾è¡¨èŠ‚ç‚¹å®šä¹‰
struct ListNode {
    int val;
    ListNode *next;
    ListNode() : val(0), next(nullptr) {}
    ListNode(int x) : val(x), next(nullptr) {}
    ListNode(int x, ListNode *next) : val(x), next(next) {}
};


class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* cur = head;
        ListNode* pre = NULL;
        ListNode* temp;
        while(cur != NULL)
        {
            temp = cur->next;
            cur->next = pre;
            pre = cur;
            cur = temp;
        }
        return pre;
    }
};

// å·¥å…·å‡½æ•°ï¼šæ‰“å°é“¾è¡?
void printList(ListNode* head) {
    ListNode* cur = head;
    while (cur != nullptr) {
        cout << cur->val << " ";
        cur = cur->next;
    }
    cout << endl;
}

// æµ‹è¯•ä¸»å‡½æ•?
int main() {
    // 1. åˆ›å»ºé“¾è¡¨ 1 -> 2 -> 3 -> 4 -> 5
    ListNode* head = new ListNode(1);
    head->next = new ListNode(2);
    head->next->next = new ListNode(3);
    head->next->next->next = new ListNode(4);
    head->next->next->next->next = new ListNode(5);

    cout << "²âÊÔÊä³ö";
    printList(head);

    Solution sol;
    ListNode* newHead = sol.reverseList(head);

    cout << "²âÊÔÊä³ö";
    printList(newHead);

    return 0;
}

File: LinkedList/fanzhuanlianbiao.java
package LinkedList;

// 链表节点类
class ListNode {
    int val;
    ListNode next;
    ListNode(int val) {
        this.val = val;
    }
}

public class fanzhuanlianbiao {

    // 你写的反转链表方法（Java版）
    public ListNode reverseList(ListNode head) {
        ListNode cur = head;
        ListNode pre = null;
        ListNode temp;

        while (cur != null) {
            temp = cur.next;
            cur.next = pre;
            pre = cur;
            cur = temp;
        }
        return pre;
    }

    // ===================== 测试代码 =====================
    public static void main(String[] args) {
        // 构建链表 1 -> 2 -> 3 -> 4 -> 5
        ListNode head = new ListNode(1);
        head.next = new ListNode(2);
        head.next.next = new ListNode(3);
        head.next.next.next = new ListNode(4);
        head.next.next.next.next = new ListNode(5);

        fanzhuanlianbiao solution = new fanzhuanlianbiao();
        ListNode newHead = solution.reverseList(head);

        // 打印反转后的链表
        ListNode cur = newHead;
        while (cur != null) {
            System.out.print(cur.val + " ");
            cur = cur.next;
        }
    }
}
